### ISSUE TITLE
TypeError: 'NoneType' object is not subscriptable

### ISSUE THREAD

**Issue by ronnyaa** [hardaker/nb2an#4]
when running  `nb-update-ansible   -c sample.yml`  and trying to use portmode.value from netbox i get a
error :: TypeError: 'NoneType' object is not subscriptable. 

sample.yml
```
test1:
  __function: foreach_create_dict
  array: interfaces
  keyname: display
  structure:
    portmode: mode.value
```
output:
```
ERROR     :     failed to call function foreach_create_dict for item test1
Traceback (most recent call last):
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/tools/update_ansible.py", line 94, in process_changes
    value = fn(dn, yaml_struct, changes[item], item)
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/plugins/update_ansible.py", line 55, in fn_foreach_create_dict
    fn_foreach_augment_dict(dn, yaml_struct, definition, item)
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/plugins/update_ansible.py", line 16, in _wrap
    return function(*args, **kwargs)
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/plugins/update_ansible.py", line 76, in fn_foreach_augment_dict
    value = dn.get(path)
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/dotnest.py", line 30, in get
    ptr = ptr[k]
TypeError: 'NoneType' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fits/.local/bin/nb-update-ansible", line 8, in <module>
    sys.exit(main())
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/tools/update_ansible.py", line 184, in main
    process_devices(nb, ansible_directory, racks=args.racks, changes=changes)
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/tools/update_ansible.py", line 159, in process_devices
    process_host(nb, name, device_yaml, changes=changes)
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/tools/update_ansible.py", line 140, in process_host
    process_changes(changes, yaml_struct, nb_data)
  File "/home/fits/.local/lib/python3.9/site-packages/nb2an/tools/update_ansible.py", line 97, in process_changes
    errors = traceback.format_exception(exp)
TypeError: format_exception() missing 2 required positional arguments: 'value' and 'tb'

```

**Comment by hardaker** [comment]
Hmmm...  a3debd6 is a fix to the error message, but not likely the error itself.

**Comment by hardaker** [comment]
doesn't that config have an error?  ie, `mode` isn't a top level netbox variable?

**Comment by ronnyaa** [comment]
mode is a interface variable, with null as the default.  if one specify the interface's  802.1Q Switching   802.1Q Mode to tagged or access it becomes a dict with a value and label key.  Is the problem that mode.value do not exist on most  (unused) interfaces?   

In my use case that   portmode: line needs to be removed as well when the mode is unset, since the https://github.com/ansible-collections/dellemc.os10  I am trying to write a host_var for is not very forgiving. 
or perhaps Ii can preload all interfaces in netbox with  802.1Q mode tagged, to ensure mode.value is always set. 



802.1Q mode unset
```
{
    "id": 487,
    "url": "https://netbox.drift.intern/api/dcim/interfaces/487/",
    "display": "Ethernet 1/1/1",
    "mode": null,
}
```

802.1Q mode tagged
```
{
    "id": 550,
    "url": "https://ds-bf-netbox.drift.intern/api/dcim/interfaces/550/",
    "display": "Ethernet 1/1/2",
    "mode": {
        "value": "tagged",
        "label": "Tagged"
    },
}
```

**Comment by hardaker** [comment]
Should be fixed in 0.9.11 [sorry for the delay -- holidays and all]
